test/strong_mode_test.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library analyzer_cli.test.strong_mode;
+
+import 'dart:io';
+import 'dart:mirrors';
+import 'package:analyzer_cli/src/driver.dart' show Driver, errorSink, outSink;
+import 'package:path/path.dart' as path;
+import 'package:unittest/unittest.dart';
+
+/// End-to-end test for --strong checking.
+///
+/// Most StrongChecker tests are in dev_compiler/test/checker/*_test.dart, but
+/// this verifies the option is working and producing extra errors as expected.
+///
+/// Generally we don't want a lot of cases here as it requires spinning up a
+/// full analysis context.
+void main() {
+  groupSep = ' | ';
+
+  group('--strong', () {
+    StringSink savedOutSink, savedErrorSink;
+    int savedExitCode;
+    setUp(() {
+      savedOutSink = outSink;
+      savedErrorSink = errorSink;
+      savedExitCode = exitCode;
+      outSink = new StringBuffer();
+      errorSink = new StringBuffer();
+    });
+    tearDown(() {
+      outSink = savedOutSink;
+      errorSink = savedErrorSink;
+      exitCode = savedExitCode;
+    });
```

Comment:
BTW: this is EXACTLY how I idiomatically use the sinks in testing the linter.  Noticing all the boilerplate makes me think we should make this a bit cleverer with a closure down the road.  But that's just gravy!